pyartifactory/objects/repository.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+import json
+import logging
+from typing import List, overload
+
+import requests
+from pydantic import TypeAdapter
+
+from pyartifactory.exception import ArtifactoryError, RepositoryAlreadyExistsError, RepositoryNotFoundError
+from pyartifactory.models import AnyRepository, AnyRepositoryResponse
+from pyartifactory.models.repository import (
+    LocalRepository,
+    LocalRepositoryResponse,
+    RemoteRepository,
+    RemoteRepositoryResponse,
+    SimpleRepository,
+    VirtualRepository,
+    VirtualRepositoryResponse,
+)
+from pyartifactory.objects.object import ArtifactoryObject
+from pyartifactory.utils import custom_encoder
+
+logger = logging.getLogger("pyartifactory")
+
+
+class ArtifactoryRepository(ArtifactoryObject):
+    """Models an artifactory repository."""
+
+    _uri = "repositories"
+
+    # Repositories operations
+    def get_repo(self, repo_name: str) -> AnyRepositoryResponse:
+        """
+        Finds repository in artifactory. Raises an exception if the repo doesn't exist.
+        :param repo_name: Name of the repository to retrieve
+        :return: Either a local, virtual or remote repository
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{repo_name}")
+            return TypeAdapter(AnyRepositoryResponse).validate_python(response.json())  # type: ignore
```

Comment:
Please remove the type ignore here.

Reply:
replaced with cast as well

Reply:
Please see my comment about using a try except method

Reply:
This try-except works if you only have two options. But in this case there are three options behind AnyRepositoryResponse. Do you want nested try-except?

Reply:
did it your way, but removed TypeAdapter